app/src/main/java/org/openobservatory/ooniprobe/common/service/ServiceUtil.java

```diff
@@ -42,72 +38,69 @@ public static void scheduleJob(Context context) {
         builder.setRequiresCharging(pm.testChargingOnly());
 
         /*
-        * Specify that this job should recur with the provided interval, not more than once per period.
-        * You have no control over when within this interval
-        * this job will be executed, only the guarantee that it will be executed at most once within this interval.
-        */
+         * Specify that this job should recur with the provided interval, not more than once per period.
+         * You have no control over when within this interval
+         * this job will be executed, only the guarantee that it will be executed at most once within this interval.
+         */
         builder.setPeriodic(60 * 60 * 1000);
         builder.setPersisted(true); //Job scheduled to work after reboot
-        
+
         //JobScheduler is specifically designed for inexact timing, so it can combine jobs from multiple apps, to try to reduce power consumption.
         JobScheduler jobScheduler = ContextCompat.getSystemService(context, JobScheduler.class);;
-        jobScheduler.schedule(builder.build());
+        if (jobScheduler != null) {
+            jobScheduler.schedule(builder.build());
+        }
     }
 
     public static void stopJob(Context context) {
         JobScheduler jobScheduler = ContextCompat.getSystemService(context, JobScheduler.class);
-        jobScheduler.cancel(id);
+        if (jobScheduler != null) {
+            jobScheduler.cancel(id);
+        }
     }
 
     public static void callCheckInAPI(Application app) {
+        app.getServiceComponent().inject(d);
+
         BatteryManager batteryManager = (BatteryManager) app.getSystemService(Context.BATTERY_SERVICE);
-        PreferenceManager pm = app.getPreferenceManager();
-        //Double checks for charging and wifi
-        if (pm.testWifiOnly() &&
```

Comment:
Where does this gode go? (until line 75) can't find it anywhere

Reply:
it's inside the GenerateAutoRunServiceSuite method shouldStart. 

I will extract it to here

Reply:
Looks great